### PR TITLE
feat(bump:deps): Add experimental homegrown update checker

### PR DESCRIPTION
**Pull request by tylerbutler**
The current implementation of the bump:deps command uses npm-check-updates to do the checking of npm and to update the package.json files. This was implemented for expediency, but the implementation is slow and does far more work than it needs to.

To address these limitations, I have added an experimental homegrown version updater. It's simpler overall, and much faster, but it needs more testing. It's behind a new CLI flag (`--updateChecker homegrown`), so the default behavior of the command won't change.

One new feature in this updater is that it will use exact dependencies when setting dependencies on pre-release versions. With prerelease, we don't want any semver range behavior -  we want exact deps.

### CODE REVIEW

File: build-tools/packages/build-cli/src/lib/package.ts [hunk 1]
```diff
@@ -681,3 +682,191 @@ export async function setPackageDependencies(
...
+ * @param depUpdateType - The constraint to use when deciding if updates are available.
+ * @param prerelease - If true, include prerelease versions as eligible to update.
+ * @param writeChanges - If true, changes will be written to the package.json files.
+ * @param log - A {@link Logger}.
+ * @returns An array of packages that had updated dependencies.
+ *
+ * @internal
+ */
+// eslint-disable-next-line max-params
+export async function npmCheckUpdates2(
```
Comment (RishhiB) [inline build-tools/packages/build-cli/src/lib/package.ts:789]: Does this need to be called npmCheckUpdates2? Is it possible to tie homegrown into it? If homegrown will change there’s no point in this then

**Review by RishhiB — APPROVED** [review]
Looks good!

File: build-tools/packages/build-cli/src/lib/package.ts (hunk 1 above)
Comment (tylerbutler) [reply build-tools/packages/build-cli/src/lib/package.ts:789]: Good point. It could totally be npmCheckaupdatesHomegrown. I'll update.

Comment (tylerbutler) [reply build-tools/packages/build-cli/src/lib/package.ts:789]: Homegrown as a string probably will change, but even so it's better to use the same name for the flags and the function.

**Pull request merged**